source/grid/grid_tools.cc

```diff
@@ -1661,13 +1669,24 @@ namespace GridTools
                     found_cell = true;
                     break;
                   }
+                const double dist = GeometryInfo<dim>::distance_to_unit_cell(p_unit);
+                if (dist < best_distance)
+                  {
+                    best_distance = dist;
+                    cell_and_position_approx.first = *cell;
+                    cell_and_position_approx.second = p_unit;
+                    approx_cell = true;
```

Comment:
I think you lack a `break` statement here. Or is your intention to keep searching through the other cells? If so, I think a comment in front of this block would be nice of the sort "The point is not inside this cell, but let us see how far outside it is and whether we want to use this cell as a backup if we can't find a cell within which it is."

I'd say that it would also be easier to read if in the block above (the one where `p_unit` is inside the cell) you'd set `approx_cell = false`, just to make clear that you are discarding the approximate cell.

Reply:
My intention is to keep searching: maybe the point is close to the edge but on another cell. Thus I want to return it only if I'm sure it's inside or it's the best approximation among all possible cases.

I believe this is also the thinking behind
`
 find_active_cell_around_point (const Mapping<dim,spacedim>  &mapping,
                                 const MeshType<dim,spacedim> &mesh,
                                 const Point<spacedim>        &p,
                                 const std::vector<bool>      &marked_vertices)
` 
but, for how that function is made, I believe currently best_distance is not really used (because, as soon as a "close enough point" is found we exit the cycle by setting found to true... so it doesn't make sense to set `best_distance = dist;` but this is for another PR).

Thank you for your comments: updating soon